### DISCUSSION TITLE
What are means of IPC nowadays?

### DISCUSSION THREAD

**Discussion by etosan** [herbstluftwm/herbstluftwm#1529] [Q&A] [answered]
Just visiting here after long hiatus. Was doing on call and got summoned to solve problem, I have now trouble sleeping, so I am revisiting my favorite projects. 

I noticed some pipe mechanism was added - does it mean we have now two IPCs at our disposal now?

**Comment by t-wissmann** [comment, marked as answer]
No, it's still only one ipc mechanism. With `herbstclient --binary-pipe` you can now send multiple commands via a single herbstclient instance.

**Comment by etosan** [comment]
This has yet not trickled down into my distribution, so I will wait for it to arrive. Question: how does this work? Does this read via stdin parsing the command lines on "\n" or one has to use argv formatted payloads (how does one know when one payload ends)?

**Reply by t-wissmann** [reply]
See the section 'binary pipe' in the herbstclient man page: https://herbstluftwm.org/herbstclient.html
